### DISCUSSION TITLE
testing HTTP request withPlainCookie() without encoding

### DISCUSSION THREAD

**Discussion by jared-f** [adonisjs/core#4649] [Help] [answered]
Hello 👋 
I am trying to make a test using HTTP client that has a simple plain cookie set. 

When I try sending a request using `withPlainCookie` I can see in the route under test that the cookie is base64 encoded json string of my plain cookie value set in the test. 

I would like to be able to disable the encoding on that cookie. 

Below are some example code: 
```js
// test

 test('It creates a taco', async ({ client, assert }) => {
    const companyName = 'Acme Inc'
    const response = await client
      .post('/tacos')
      .withPlainCookie('sauce','spicy')
      .form({ toppings: [ 'cilantro'] })
      .withCsrfToken()
    response.assertStatus(200)
    const taco = await Taco.findBy({ toppings: 'cilantro' })
    assert.ok(taco, 'It should make a tacos')
  })
  ```
  
  In the route when I inspect `ctx.request.plainCookie('sauce')` I have a string like this: 
  `eyJtZXNzYWdlIjogInNwaWN5In0=`

which is a json object like this: 
```json
{ "message": "spicy"}
```

Please advise how I can pass a plaincookie without the json / base64. 

when I test with my own client I can send the plain cookie and the routes work. The issue is only under test using `withPlainCookie`. 

thanks

**Comment by thetutlage** [comment, marked as answer]
You can read the cookie as follows via request.

```ts
ctx.request.plainCookie('sauce', {
  encoded: false
})
```

But I agree, it will be nice to allow non-encoded cookies via tests too. Can you please file an issue in https://github.com/adonisjs/http-server repo for the same and I will look into it

**Reply by jared-f** [reply]
Thanks @thetutlage !

Agreed, it works as expected to read the cookie with `plainCookie` outside of testing. Doing API Client functional testing it does not. 
I will file the issue.
